### PR TITLE
Move to GH actions

### PR DESCRIPTION
**Pull request by ennru**

### CODE REVIEW

File: .github/workflows/check-build-test.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      JAVA_OPTS: -Xms2G -Xmx2G -Xss2M -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with: # https://github.com/olafurpg/setup-scala#faster-checkout-of-big-repos
+          fetch-depth: 100
+
+      - name: Fetch tags
+        run: git fetch --depth=100 origin +refs/tags/*:refs/tags/*
```
Comment (patriknw) [inline .github/workflows/check-build-test.yml:83]: does this merge with main? in another repo I have:
```
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Checkout GitHub merge
        if: github.event.pull_request
        run: |-
          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
          git checkout scratch
```

File: .github/workflows/check-build-test.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      - name: "Create all API docs and create site with Paradox"
+        run: sbt "root/doc; docs/makeSite"
+
+      - name: Run Link Validator
+        run: cs launch net.runne::site-link-validator:0.2.2 -- project/link-validator.conf
+
+  test:
+    name: Test with the Spanner emulator
+    runs-on: ubuntu-20.04
```
Comment (patriknw) [inline .github/workflows/check-build-test.yml:71]: I think all of them should have
```
if: github.repository == 'akka/akka-persistence-spanner'
```

File: .github/workflows/check-build-test.yml (hunk 1 above)
Comment (ennru) [reply .github/workflows/check-build-test.yml]: Right, I changed the checkout for tests to use the merge.

File: .github/workflows/check-build-test.yml (hunk 2 above)
Comment (ennru) [reply .github/workflows/check-build-test.yml:71]: All but releasing should be OK to run on forks, as well.

Comment (patriknw) [reply .github/workflows/check-build-test.yml:71]: > All but releasing should be OK to run on forks, as well.

Fine, but probably surprising for contributors and not very environmental friendly.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**